### PR TITLE
Grover with sampler tutorial

### PR DESCRIPTION
**Pull request by HuangJunye**
### Summary

Grover's algorithm with Sampler program tutorial. Still a work in progress but any feedback in welcome.

### CODE REVIEW

**Review by frankharkins — COMMENTED** [review]
I got an error when running the jobs which I mentioned above, can provide more info if needed. I also left some minor suggestions but I know this is still WIP so feel free to ignore. Otherwise, looks great.

File: docs/tutorials/grover_with_sampler.ipynb [hunk 1]
```diff
@@ -0,0 +1,409 @@
...
+    }
+   ],
+   "source": [
+    "from qiskit_ibm_runtime import IBMSampler\n",
+    "\n",
+    "sampler_factory = IBMSampler(service=service)\n",
+    "\n",
+    "with sampler_factory(circuits=grover_circuits) as sampler:\n",
+    "    result = sampler(circuit_indices=[0,1], shots=1000)\n",
+    "    print(result)"
```
[the remark this answers is not in the document]
Comment (HuangJunye) [reply docs/tutorials/grover_with_sampler.ipynb:302]: That's weird. I have never see this error before. What's your OS, python and Qiskit versions?

File: docs/tutorials/grover_with_sampler.ipynb [hunk 2]
```diff
@@ -0,0 +1,409 @@
...
+     },
+     "execution_count": 11,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "# Let's look at the results\n",
+    "from qiskit.tools.visualization import plot_histogram\n",
+    "plot_histogram(result.quasi_dists, legend=['1 iteration', '2 iterations'])"
```
[the remark this answers is not in the document]
Comment (HuangJunye) [reply docs/tutorials/grover_with_sampler.ipynb:341]: I agree this is probably related to the execution error above. Let me take a look.

File: docs/tutorials/grover_with_sampler.ipynb [hunk 3]
```diff
@@ -0,0 +1,409 @@
...
+     "shell.execute_reply": "2022-03-24T09:31:02.691512Z",
+     "shell.execute_reply.started": "2022-03-24T09:30:57.712543Z"
+    },
+    "tags": []
+   },
+   "outputs": [],
+   "source": [
+    "from qiskit_ibm_runtime import IBMRuntimeService\n",
+    "\n",
+    "service = IBMRuntimeService()"
```
[the remark this answers is not in the document]
Comment (HuangJunye) [reply docs/tutorials/grover_with_sampler.ipynb]: I removed this code in step 1.

File: docs/tutorials/grover_with_sampler.ipynb [hunk 4]
```diff
@@ -0,0 +1,523 @@
...
+    "secret_string = format(secret, '03b')  # format the owner in 3-bit string\n",
+    "oracle = Statevector.from_label(secret_string)  # let the oracle know the owner"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "ab38ee5a-a0f4-4dc7-bdef-a733cf3ce153",
+   "metadata": {},
+   "source": [
+    "Once we have the oracle, we can define the unstractured search problem using the `AmplificationProblem` class in Qiskit. "
```
Comment (rathishcholarajan) [inline docs/tutorials/grover_with_sampler.ipynb:82]:
```suggestion
    "Once we have the oracle, we can define the unstructured search problem using the `AmplificationProblem` class in Qiskit. "
```

File: docs/tutorials/grover_with_sampler.ipynb [hunk 5]
```diff
@@ -0,0 +1,523 @@
...
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "ed5a7c5f-5f21-40d4-b4d1-37f8bb46337e",
+   "metadata": {},
+   "source": [
+    "## 4. Submit the circuits to IBM Cloud\n",
+    "\n",
+    "Now that the Grover's circuits are created, let's submit them to a quantum computer on IBM Cloud by using the Sampler program."
```
Comment (rathishcholarajan) [inline docs/tutorials/grover_with_sampler.ipynb:226]: Gentle reminder to make this generic as well, for IBM Cloud or IBM Quantum when fixing Step 1.

**Comment by HuangJunye** [comment]
@rathishcholarajan Thanks for the review. I have committed the suggested changes. I'll make the changes on the mention of IBM Cloud a bit later.

File: docs/tutorials/grover_with_sampler.ipynb (hunk 5 above)
Comment (HuangJunye) [reply docs/tutorials/grover_with_sampler.ipynb]: I change 224 to "Submit the circuits to a quantum computer on the cloud" and 226 to "a quantum computer on the cloud". What do you think?

File: docs/tutorials/grover_with_sampler.ipynb [hunk 6]
```diff
@@ -0,0 +1,523 @@
...
+     "shell.execute_reply": "2022-04-07T13:47:27.506544Z",
+     "shell.execute_reply.started": "2022-04-07T13:47:27.503446Z"
+    },
+    "tags": []
+   },
+   "outputs": [
+    {
+     "data": {
+      "text/plain": [
+       "'0.3.0'"
```
Comment (rathishcholarajan) [inline docs/tutorials/grover_with_sampler.ipynb:433]:
```suggestion
       "'0.4.0'"
```

**Review by rathishcholarajan — APPROVED** [review]
LGTM! Thanks a lot @HuangJunye!

**Pull request merged**